packages/core/src/components/graphs/gauge.ts

```diff
@@ -0,0 +1,356 @@
+			.text(d => `${numberFormatter(d)}%`);
+
+		// Add the caret for the delta number
+		const {
+			width: deltaNumberWidth
+		} = DOMUtils.getSVGElementSize(DOMUtils.appendOrSelect(svg, "g.gauge-delta"), { useBBox: true });
+
+		const deltaArrow = deltaGroup.selectAll("svg.gauge-delta-arrow")
+			.data(delta != null ? [delta] : []);
+
+		deltaArrow
+			.enter()
+			.append("svg")
+			.merge(deltaArrow)
+			.attr("class", "gauge-delta-arrow")
+			.attr("x", -arrowSize(radius) - deltaNumberWidth / 2)
+			.attr("y", -arrowSize(radius) / 2 - deltaFontSize(radius) * 0.35)
+			.attr("width", arrowSize(radius))
+			.attr("height", arrowSize(radius))
+			.attr("viewBox", "0 0 16 16");
+
+		// Needed to correctly size SVG in Firefox
+		DOMUtils.appendOrSelect(deltaArrow, "rect.gauge-delta-arrow-backdrop")
+			.attr("width", "16")
+			.attr("height", "16")
+			.attr("fill", "none");
+
+		// Draw the arrow with status
+		const status = Tools.getProperty(options, "gauge", "status");
+		DOMUtils.appendOrSelect(deltaArrow, "polygon.gauge-delta-arrow-polygon")
+			.classed(`status--${status}`, status != null)
+			.attr("fill", () => status == null ? "currentColor" : null)
+			.attr("points", self.getArrow(delta));
+
+		deltaArrow.exit().remove();
+		deltaNumber.exit().remove();
+	}
+
+	getInnerRadius() {
+		// Compute the outer radius needed
+		const radius = this.computeRadius();
+		const arcWidth = Tools.getProperty(this.model.getOptions(), "gauge", "arcWidth");
+		return radius - arcWidth;
+	}
+
+	addEventListeners() {
+		const self = this;
+		this.parent
+			.selectAll("path.arc")
+			.on("mouseover", function (datum) {
+				// Dispatch mouse event
+				self.services.events.dispatchEvent(Events.Gauge.ARC_MOUSEOVER, {
+					element: select(this),
+					datum
+				});
+			})
+			.on("mousemove", function (datum) {
+				const hoveredElement = select(this);
+
+				// Dispatch mouse event
+				self.services.events.dispatchEvent(Events.Gauge.ARC_MOUSEMOVE, {
+					element: hoveredElement,
+					datum
+				});
+
+				// Show tooltip
+				self.services.events.dispatchEvent(Events.Tooltip.SHOW, {
+					hoveredElement,
+					type: TooltipTypes.DATAPOINT
+				});
+			})
+			.on("click", function (datum) {
+				// Dispatch mouse event
+				self.services.events.dispatchEvent(Events.Gauge.ARC_CLICK, {
+					element: select(this),
+					datum
+				});
+			})
+			.on("mouseout", function (datum) {
+				const hoveredElement = select(this);
+
+				// Dispatch mouse event
+				self.services.events.dispatchEvent(Events.Gauge.ARC_MOUSEOUT, {
+					element: hoveredElement,
+					datum
+				});
+
+				// Hide tooltip
+				self.services.events.dispatchEvent(Events.Tooltip.HIDE, {
+					hoveredElement
+				});
+			});
+	}
+
+	// Helper functions
+	protected computeRadius() {
+		const options = this.model.getOptions();
+		const arcType = Tools.getProperty(options, "gauge", "type");
+
+		const { width, height } = DOMUtils.getSVGElementSize(this.parent, {
+			useAttrs: true
+		});
```

Comment:
Can this be one line? You mentioned that even single property objects shouldn't be placed on one line so if not the places you have `DOMUtils.getSVGElementSize(DOMUtils.appendOrSelect(svg, "text.gauge-value-number"), { useBBox: true })` should be changed

Reply:
it doesnt object with a max-line flag but prettier puts it back like this